sync-core/src/main/java/com/cloudant/sync/datastore/BasicDatastore.java

```diff
@@ -791,53 +805,69 @@ private BasicDocumentRevision deleteDocumentInQueue(SQLDatabase db, final String
 
         CouchUtils.validateRevisionId(prevRevId);
 
-                    BasicDocumentRevision deletedRevision = null;
+        BasicDocumentRevision deletedRevision = null;
+            BasicDocumentRevision preRevision;
+        try {
+            preRevision = getDocumentInQueue(db, docId, prevRevId);
+        }  catch (AttachmentException e){
+            throw new DocumentNotFoundException(e);
+        }
```

Comment:
I can't parse that sentence, " to stop the datastore creating a document when it shouldn't error" -- I think altering the use of commas might help?